### ISSUE TITLE
Bug: Problems with —blankLinesBetweenScopes and --self insert

### ISSUE THREAD

**Issue by nallick** [nicklockwood/SwiftFormat#407]
**Steps to reproduce:**
In an empty directory add Test.swift with:
```
struct S {
	var p: String { return "" }
	func m1() {
		let s = p
		m2(s)
	}
	func m2(_: String) {}
}
```

From the same directory run:
`swiftformat --self insert ./Test.swift
`

**Expected results:**
```
struct S {
	var p: String { return "" }

	func m1() {
		let s = self.p
		self.m2(s)
	}

	func m2(_: String) {}
}
```

**Actual results:**
```
struct S {
	var p: String { return "" }
	func m1() {
		let s = p
		m2(s)
	}

	func m2(_: String) {}
}
```

**Notes:**
There is a missing blank line before m1() and self is not inserted when referencing the property or method. The "let" seems to be inhibiting the addition of self on its line and the following line. If the trailing "}" in the var definition is moved to the following line a blank line is added after it. I would also expect a line between a non-computed var and the following function (e.g., var p: String = ""), but that doesn't happen either.

**Comment by nicklockwood** [comment]
@nallick thanks for reporting. I'll try to get these fixed for the next release.

**Comment by nicklockwood** [comment]
@nallick the self insertion bug is fixed in version 0.40.6.

**Comment by nallick** [comment]
Thank you. I'll check it out.
